File: Exercicios-LPA-main/Aula 24/ex1/ex1.py
import argparse
from typing import List, Optional

def escrever_parametros(nome_arquivo: str, params: List[str]) -> None:
    """
    Escreve uma lista de parâmetros em um arquivo, cada um em sua própria linha.
    
    Args:
        nome_arquivo: Nome do arquivo onde os parâmetros serão escritos
        params: Lista de strings contendo os parâmetros
        
    Raises:
        IOError: Se ocorrer erro ao abrir ou escrever no arquivo
    """
    try:
        # Usando 'with' para garantir que o arquivo seja fechado corretamente
        with open(nome_arquivo, 'w', encoding='utf-8') as arquivo:
            # Escrevemos cada parâmetro seguido por uma nova linha
            for param in params:
                arquivo.write(f"{param}\n")
        print(f"Parâmetros escritos com sucesso no arquivo '{nome_arquivo}'")
    except IOError as e:
        print(f"Erro ao escrever no arquivo: {str(e)}")

def main() -> None:
    """Função principal que processa os argumentos da linha de comando."""
    parser = argparse.ArgumentParser(
        description="Programa para gravar parâmetros em um arquivo",
        formatter_class=argparse.ArgumentDefaultsHelpFormatter
    )
    
    parser.add_argument(
        "arquivo",
        help="Nome do arquivo onde os parâmetros serão gravados"
    )
    
    parser.add_argument(
        "params",
        nargs="+",
        help="Lista de parâmetros a serem gravados"
    )
    
    args = parser.parse_args()
    escrever_parametros(args.arquivo, args.params)

if __name__ == "__main__":
    main()

File: Exercicios-LPA-main/Aula 24/ex1/ex1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Função para escrever parâmetros em um arquivo
static int escrever_parametros(const char *nome_arquivo, const char **params, size_t num_params) {
    FILE *arquivo;
    
    // Tentativa de abrir o arquivo em modo escrita
    arquivo = fopen(nome_arquivo, "w");
    if (arquivo == NULL) {
        fprintf(stderr, "Erro ao abrir arquivo '%s': %s\n", 
                nome_arquivo, strerror(errno));
        return 0;
    }

    // Escreve cada parâmetro em uma linha separada
    for (size_t i = 0; i < num_params; i++) {
        if (fprintf(arquivo, "%s\n", params[i]) < 0) {
            fprintf(stderr, "Erro ao escrever no arquivo\n");
            fclose(arquivo);
            return 0;
        }
    }

    fclose(arquivo);
    printf("Parâmetros escritos com sucesso no arquivo '%s'\n", nome_arquivo);
    return 1;
}

int main(int argc, char *argv[]) {
    int i;
    
    // Verificação de argumentos mínimos necessários
    if (argc < 3) {
        fprintf(stderr, "Uso: %s <nome_arquivo> <parametro1> [parametro2 ...]\n", argv[0]);
        return EXIT_FAILURE;
    }

    // Chamada da função principal com os parâmetros fornecidos
    const char **params = &argv[2];
    size_t num_params = argc - 2;

    return !escrever_parametros(argv[1], params, num_params);
}